opacus/tests/grad_samples/embedding_norm_sample_test.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from opacus.grad_sample import embedding_norm_sample
+import torch
+import torch.nn as nn
+
+
+class TestComputeEmbeddingNormSample(unittest.TestCase):
+
+    def test_compute_embedding_norm_sample(self):
+        # Define the embedding layer
+        embedding_dim = 1
+        vocab_size = 3
+        embedding_layer = nn.Embedding(vocab_size, embedding_dim)
+
+        # Manually set weights for the embedding layer for testing
+        embedding_layer.weight = nn.Parameter(
+            torch.tensor([[0.1], [0.2], [0.3]], dtype=torch.float32)
+        )
+
+        # Example input ids (activations). Shape: [3, 2]
+        input_ids = torch.tensor([[1, 1], [2, 0], [2, 0]], dtype=torch.long)
+
+        # Example gradients with respect to the embedding output (backprops).
+        # Shape: [6, 1]
```

Comment:
Nitpicking: we should have size [3,2,1] for backprops, correct? [6, 1] is for grad_values.